docs/madr/decisions/071-transparent-proxy-config-map-handling-outside-the-control-plane.md

```diff
@@ -0,0 +1,173 @@
+
+This document proposes removing that requirement for setups that don’t use `kuma-cni`. Instead of having the control plane build the full config, each data plane component (`kuma-init` and `kuma-sidecar`) will build its own.
+
+## Out of Scope
+
+`kuma-cni` is deployed in the `kube-system` namespace. Without expanding its permissions to access ConfigMaps from other namespaces, it cannot support custom ConfigMaps specified at the workload level. With the proposed changes, where config building moves outside of the control plane, `kuma-cni` will no longer be able to consume workload-specific ConfigMaps at all. This document **_WON'T_** cover the changes required to support such functionality in `kuma-cni`. As a result, the transparent proxy configuration via ConfigMaps will be incomplete and backward incompatible when using `kuma-cni`.
+
+## Problems
+
+### Control plane needs the config to build the Dataplane object
+
+There are two control plane components that require transparent proxy configuration:
+
+- **Sidecar Injector** - injects `kuma-sidecar` and `kuma-init` containers into Pods that are part of the mesh
+- **Pod Reconciler** - creates and updates the `Dataplane` object based on the Pod
+
+The `Dataplane` object includes transparent proxy settings like `ipFamilyMode`, `redirectPortInbound`, and `redirectPortOutbound`. These are used by the control plane to generate the xDS config delivered to `kuma-sidecar`.
+
+Sidecar injection and Dataplane generation are handled by separate components, with no shared state. This means there is no direct way to pass configuration between them.
+
+### Each component gets config in a different way
+
+- `kuma-init`: gets full config as a CLI argument
+- `kuma-sidecar`: gets config as environment variables (`KUMA_DNS_ENABLED`, `KUMA_DNS_CORE_DNS_PORT`)
+
+This inconsistency makes it hard to track where values are coming from.
+
+## Solution
+
+We stop using the transparent proxy config fields in the Dataplane object and mark them as deprecated. These fields will no longer be relied on by `kuma-sidecar` in Kubernetes deployments. Instead, the same information will be passed as metadata during the xDS connection, using values from the transparent proxy config passed via a CLI flag.
+
+`kuma-dp` will support a `--transparent-proxy` flag, and `kumactl install transparent-proxy` will support a unified `--config` flag. Both flags can accept one of the following values:
+
+- No value: uses the default transparent proxy config (allows running `kuma-dp` without enabling the proxy)
+- A path to a single config file
+- A path to a directory with one or more config files, read in alphabetical order (later files override earlier ones)
+- A raw YAML string with the config
```

Comment:
Shouldn't it stick with the more std "-" and read stdin. 
You could then combine: 

```
echo 'baseconf' | kumactl install transparent-proxy --transparent-proxy-config /var/conf.yaml,-
```